compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -49,7 +49,8 @@ class Compiler {
       List(new LinkAll),            // Reload compilation units from TASTY for library code (if needed)
       List(new FirstTransform,      // Some transformations to put trees into a canonical form
            new CheckReentrant,      // Internal use only: Check that compiled program has no data races involving global vars
-           new ElimJavaPackages),   // Eliminate syntactic references to Java packages
+           new ElimPackagePrefixes, // Eliminate syntactic references to Java packages
+           new ReifyQuotes),        // Eliminate syntactic references to Java packages
```

Comment:
We might need to do this before `FirstTranform`, when we will pickle the contents of the quotes we should probably give that pickler a tree that is in the same shape as the tree was in `Pickler`.

Reply:
FirstTransforms does some simplifications. It eliminates NamedArgs and Imports and replaces  all trees representing types by TypeTree nodes. This is arguable good, since it simplifies the trees to reify quite a bit. The reason we keep the more complicated trees in the TASTY is for the IDE, so that all elements of source files are represented. But for macros and staging that should not be necessary.

Reply:
On second thought, I now see you were right. It does become easier if we reify exactly in the pickled format.